File: src/main/java/com/pairingplanet/pairing_planet/dto/log_post/LogPostDetailResponseDto.java
package com.pairingplanet.pairing_planet.dto.log_post;

import com.pairingplanet.pairing_planet.dto.image.ImageResponseDto; // [추가]
import com.pairingplanet.pairing_planet.dto.recipe.RecipeSummaryDto;

import java.time.Instant;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

public record LogPostDetailResponseDto(
        UUID publicId,
        String title,
        String content,
        String outcome,  // SUCCESS, PARTIAL, FAILED
        List<ImageResponseDto> images, // [수정] List<String> -> List<ImageResponseDto>
        RecipeSummaryDto linkedRecipe,
        Instant createdAt
) {}

File: lib/data/models/log_post/log_post_detail_response_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pairing_planet2_frontend/data/models/image/image_response_dto.dart';
import 'package:pairing_planet2_frontend/data/models/recipe/recipe_summary_dto.dart';

part 'log_post_detail_response_dto.g.dart';

@JsonSerializable()
class LogPostDetailResponseDto {
  final String publicId;
  final String title;
  final String content;
  final int rating;
  final List<ImageResponseDto> images; // [수정] String -> ImageResponseDto
  final RecipeSummaryDto? linkedRecipe;

  LogPostDetailResponseDto({
    required this.publicId,
    required this.title,
    required this.content,
    required this.rating,
    required this.images,
    this.linkedRecipe,
  });

  factory LogPostDetailResponseDto.fromJson(Map<String, dynamic> json) =>
      _$LogPostDetailResponseDtoFromJson(json);
  Map<String, dynamic> toJson() => _$LogPostDetailResponseDtoToJson(this);
}
